src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,175 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/** Duke class to encapsulate the behaviour of a task manager */
+public class Duke {
+    private static final String SAVE_PATH = "./src/data/SaveData.txt";
+    private static enum AcceptedCommands {
+        TODO,
+        EVENT,
+        DEADLINE,
+        LIST,
+        DONE,
+        BYE,
+        DELETE,
+        CLEAR,
+        HELLO,
+        FIND,
+    }
+    public static Scanner scan = new Scanner(System.in);
+    private Storage storage;
+    private TaskList tasks;
+
+    public Duke(String filepath) {
+        storage = new Storage(filepath);
+        tasks = storage.loadTask();
+    }
+
+    /** Driver method for Duke */
+    private void run(String userInput) {
+        while (!userInput.equals("bye")) {
+            try {
+                String command = Parser.getCommand(userInput);
+                String details = Parser.getDetails(userInput);
+
+                // Catch illegal commands
+                checkIllegalCommand(command);
+                // Catch missing content
+                checkMissingArgument(command, details);
+                // Check indexing out of bounds
+                checkExistingTask(command, details, tasks.length());
+                // Check illegal argument
+                checkIllegalArgument(command, details);
+
+                // Decide the actions
+                String[] description;
+                Task taskToUpdate;
+                switch (command) {
+                    case "list":
```

Comment:
Should the indentation for "case" be 1 tab less?

Reply:
Agree, the coding standard states "there is no indentation for case clauses."